hydra-node/src/Hydra/Ledger/Cardano/Orphans.hs

```diff
@@ -0,0 +1,563 @@
+      Left err -> fail $ "failed to decode from base16: " <> show err
+      Right bs' -> case decodeAnnotator "Redeemers" fromCBOR (fromStrict bs') of
+        Left err -> fail $ show err
+        Right v -> pure v
+
+instance ToCBOR (Ledger.Alonzo.Redeemers era) => ToJSON (Ledger.Alonzo.Redeemers era) where
+  toJSON = String . decodeUtf8 . Base16.encode . serialize'
+
+--
+-- SafeHash
+--
+
+instance Crypto crypto => ToJSONKey (Ledger.SafeHash crypto any) where
+  toJSONKey = toJSONKeyText safeHashToText
+
+safeHashToText ::
+  Ledger.SafeHash crypto any ->
+  Text
+safeHashToText =
+  decodeUtf8 . Base16.encode . Crypto.hashToBytes . Ledger.extractHash
+
+instance Crypto crypto => FromJSON (Ledger.SafeHash crypto any) where
+  parseJSON = withText "SafeHash" safeHashFromText
+
+instance Crypto crypto => FromJSONKey (Ledger.SafeHash crypto any) where
+  fromJSONKey = FromJSONKeyTextParser safeHashFromText
+
+safeHashFromText ::
+  (Crypto crypto, MonadFail m) =>
+  Text ->
+  m (Ledger.SafeHash crypto any)
+safeHashFromText t =
+  case Crypto.hashFromTextAsHex t of
+    Nothing -> fail "failed to decode from base16."
+    Just h -> pure $ Ledger.unsafeMakeSafeHash h
+
+--
+-- Script
+--
+
+instance
+  ( ToCBOR (Ledger.Alonzo.Script era)
+  ) =>
+  ToJSON (Ledger.Alonzo.Script era)
+  where
+  toJSON = String . decodeUtf8 . Base16.encode . serialize'
+
+instance
+  ( Crypto (Ledger.Crypto era)
+  , Typeable era
+  ) =>
+  FromJSON (Ledger.Alonzo.Script era)
+  where
+  parseJSON = withText "Script" $ \t ->
+    case Base16.decode $ encodeUtf8 t of
+      Left err -> fail $ "failed to decode from base16: " <> show err
+      Right bs' -> case decodeAnnotator "Script" fromCBOR (fromStrict bs') of
+        Left err -> fail $ show err
+        Right v -> pure v
+
+--
+-- ScriptHash
+--
+
+instance Crypto crypto => ToJSONKey (Ledger.ScriptHash crypto) where
+  toJSONKey = toJSONKeyText $ \(Ledger.ScriptHash h) ->
+    decodeUtf8 $ Base16.encode (Crypto.hashToBytes h)
+
+instance Crypto crypto => FromJSONKey (Ledger.ScriptHash crypto) where
+  fromJSONKey = FromJSONKeyTextParser $ \t ->
+    case Crypto.hashFromTextAsHex t of
+      Nothing -> fail "failed to decode from base16."
+      Just h -> pure $ Ledger.ScriptHash h
+
+--
+-- Timelock
+--
+
+instance ToJSON (Ledger.Mary.Timelock StandardCrypto) where
+  toJSON = String . decodeUtf8 . Base16.encode . serialize'
+
+instance FromJSON (Ledger.Mary.Timelock StandardCrypto) where
+  parseJSON = withText "Timelock" $ \t ->
+    case Base16.decode $ encodeUtf8 t of
+      Left e -> fail $ "failed to decode from base16: " <> show e
+      Right bs' -> case decodeAnnotator "Timelock" fromCBOR (fromStrict bs') of
+        Left err -> fail $ show err
+        Right v -> pure v
+
+--
+-- TxBody
+--
+
+instance ToJSON (Ledger.Alonzo.TxBody StandardAlonzo) where
+  toJSON b =
+    object $
+      mconcat
+        [ onlyIf (const True) "inputs" (Set.map fromShelleyTxIn (Ledger.Alonzo.inputs' b))
```

Comment:
Why the `onlyIf`, if this is always included?

Reply:
Consistency with the rest. It reads better.